File: C++/프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> peoples(n,1);
    for(int i : lost) {
        peoples[i-1] -= 1;
    }
    for(int i : reserve) {
        peoples[i-1] += 1;
    }
    for(int i = 0; i < n; i++) {
        if(peoples[i] == 2) {
            if(i-1 >= 0 and peoples[i-1] == 0) {
                peoples[i] = 1;
                peoples[i-1] = 1;
            }
            else if(i+1 < n and peoples[i+1] == 0) {
                peoples[i] = 1;
                peoples[i+1] = 1;
            }
        }
    }
    int ans = 0;
    for(int i : peoples) {
        ans += (i ? 1 : 0); 
        // cout << i << ' ';
    }
    return ans;
}

File: Python3/프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    have = [1]*(n+1);
    for i in lost:
        have[i] -= 1
    for i in reserve:
        have[i] += 1
    for i in range(1,n+1):
        if have[i] == 0:
            if i-1 > 0 and have[i-1] == 2:
                have[i] = 1
                have[i-1] = 1
            elif i+1 <= n and have[i+1] == 2:
                have[i] = 1
                have[i+1] = 1
    return sum([1 for i in have if i != 0]) -1